src/jsep.js

```diff
@@ -249,23 +251,26 @@ export class Jsep {
 		|| ch === Jsep.CR_CODE) {
 			ch = this.expr.charCodeAt(++this.index);
 		}
+		Jsep.hooks.run('gobble-spaces', this);
 	}
 
 	/**
 	 * Top-level method to parse all expressions and returns compound or single node
 	 * @returns {jsep.Expression}
 	 */
 	parse() {
-		const nodes = this.gobbleExpressions();
+		Jsep.hooks.run('before-all', this);
+		this.nodes = this.gobbleExpressions();
+		Jsep.hooks.run('after-all', this);
```

Comment:
could potentially pass in the nodes[] array. Hooks wouldn't be able to reassign to it (but would be able to push/pop to manipulate the array)

Reply:
You're passing in `this`, so you're passing that as well. Not sure what change you're proposing?

Reply:
I changed this to run with just env.node. So now all uses would just be accessing `env.node`.